### PR TITLE
Support OAuth authentication for Big Query

### LINKED ISSUE #420: Support OAuth authentication for Big Query
Hi there!

Currently, cosmos only supports two authentication mechanisms:

1. service account key file and 
2. service account key json

DBT supports `oauth via gcloud` authentication. See documentation [here](https://docs.getdbt.com/docs/core/connect-data-platform/bigquery-setup#oauth-via-gcloud).

As part of this issue, I request the support for `oauth via gcloud` authentication method.

I will open a PR for the same and if everything looks good, we can hopefully merge it into `main` and make it a part of the next release.

Thanks!

### PR DESCRIPTION
**Pull request by MonideepDe**
## Description

Support OAuth authentication for Big Query

## Related Issue(s)
closes #420 

## Breaking Change?

None

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cosmos/profiles/bigquery/oauth.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+"Maps Airflow GCP connections to dbt BigQuery profiles that uses oauth via gcloud, if they don't use key file or JSON."
+from typing import Any
```
Comment (jlaneve) [inline cosmos/profiles/bigquery/oauth.py:2]:
```suggestion
"Maps Airflow GCP connections to dbt BigQuery profiles that uses oauth via gcloud, if they don't use key file or JSON."
from __future__ import annotations

from typing import Any
```

Comment (jlaneve) [reply cosmos/profiles/bigquery/oauth.py:2]: I think adding `from __future__ import annotations` should fix this

Comment (MonideepDe) [reply cosmos/profiles/bigquery/oauth.py]: Thanks for the suggestion. I have added this 😄

**Comment by MonideepDe** [comment]
@jlaneve The tests are failing with:
```
[2023-08-02 09:29:21,675] {graph.py:185} INFO - Skipping line:   Env var required but not provided: 'POSTGRES_HOST'
```
This PR does not alter anything related to `POSTGRES_HOST`. I can see the same issue in [another PR's check](https://github.com/astronomer/astronomer-cosmos/actions/runs/5731463182/job/15535676049?pr=429#step:6:382) as well.
I can see the errors when I run the tests against the `main` branch on my local as well 😞 

Are you aware of a test bug of some sorts that has creeped in and if there is a PR already in the work for the same?

Thanks!

**Comment by jlaneve** [comment]
> @jlaneve The tests are failing with:
> 
> ```
> [2023-08-02 09:29:21,675] {graph.py:185} INFO - Skipping line:   Env var required but not provided: 'POSTGRES_HOST'
> ```
> 
> This PR does not alter anything related to `POSTGRES_HOST`. I can see the same issue in [another PR's check](https://github.com/astronomer/astronomer-cosmos/actions/runs/5731463182/job/15535676049?pr=429#step:6:382) as well. I can see the errors when I run the tests against the `main` branch on my local as well 😞
> 
> Are you aware of a test bug of some sorts that has creeped in and if there is a PR already in the work for the same?
> 
> I hope I am missing something obvious 😢
> 
> Thanks!

Unfortunately this is a known issue when there are PRs from forked repos because CI secrets don’t get set. As long as the unit tests and pre-commit checks pass, we can merge this! It looks like there’s just one minor issue with the pre-commit checks - one of the lines is too long

**Review by jlaneve — APPROVED** [review]
this looks great, thank you so much @MonideepDe!

**Pull request merged**